src/mds/agg_mds/commons.py

```diff
@@ -10,25 +10,198 @@
 class ColumnsToFields:
     """
     A more complex mapping object for mapping column names to MDS fields
-    allows to explictly mark a field as missing, a default value and it's resources type
+    allows to explicitly mark a field as missing, a default value and it's resources type
     """
 
     name: str
     missing: bool = False
     default: str = ""
     type: str = "string"
 
+    def get_value(self, info: dict):
+        return info.get(self.name, self.default)
+
+
+@dataclass_json
+@dataclass
+class FieldAggregation:
+    """
+    Provides a description of what fields to compute summary information.
+    The default assumes computing the sum of the field, assuming it is a number
+    the functions supported are: sum and count
+    """
+
+    type: str = "number"
+    function: str = "sum"
+    chart: str = "text"
+
+
+FieldDefinition = TypeVar("FieldDefinition")
+
+
+def string_to_array(s: str) -> Optional[List[str]]:
+    if s == "":
+        return []
+    return [s]
+
+
+def array_to_string(arr: Optional[list]) -> Optional[str]:
+    if arr is None:
+        return None
+    return "".join(arr)
+
+
+def string_to_integer(s: str) -> int:
+    return int(s) if s.isnumeric() else None
+
+
+def string_to_number(s: str) -> Optional[float]:
+    try:
+        return float(s)
+    except ValueError:
+        return None
+
+
+def string_to_dict(s: str) -> Optional[Dict[Any, Any]]:
+    try:
+        return json.loads(s)
+    except json.JSONDecodeError:
+        return None
+
+
+def dict_to_array(d: dict) -> List[Dict[Any, Any]]:
+    return [d]
```

Comment:
probably a good idea to log a warning when any of these conversions are failing?

Reply:
Added